src/core/net/nat64_translator.cpp

```diff
@@ -99,7 +99,8 @@ Error Translator::SendMessage(Message &aMessage)
 
     VerifyOrExit(result == kForward);
 
-    error = Get<Ip6::Ip6>().SendRaw(aMessage, !OPENTHREAD_CONFIG_IP6_ALLOW_LOOP_BACK_HOST_DATAGRAMS);
+    aMessage.SetOrigin(Message::Origin::kFromHostUntrusted);
```

Comment:
Minor considerations: The message origin should set already, since the NAT64 translator will reuse the same message object. I would prefer add a unit test for `TranslateToIp6` that it does not touch the MessageOrigin field.

If we want to preserve the pattern to set the message origin before calling Ip6.SendRaw, I would suggest:

```cpp
...
Message::Origin messageOrigin = aMessage.GetOrigin();
Result result = TranslateToIp6(aMessage);

VerifyOrExit(result == kForward);

aMessage.SetOrigin(messageOrigin);
...
```

Then for NAT64, we have a single point setting the MessageOrigin.